src/components/CalcStructure.js

```diff
@@ -1,42 +1,152 @@
-const CalcStructure = () => (
-  <div>
-    <div className="output-screen">
-      0
+import { useState } from 'react';
+import calculate from '../logic/calculate';
+
+const CalcStructure = () => {
+  const [result, setResult] = useState('0');
+  const [calculatorData, setCalculatorData] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  const handleButtonClick = (event) => {
+    const buttonValue = event.target.textContent;
+    const updatedCalculatorData = calculate(calculatorData, buttonValue);
+    setCalculatorData(updatedCalculatorData);
+
+    // Update the result display
+    if (updatedCalculatorData.operation) {
+      if (updatedCalculatorData.next) {
+        setResult(`${updatedCalculatorData.total.toString()
+        }  ${updatedCalculatorData.operation.toString()}  ${updatedCalculatorData.next.toString()}`);
+      } else {
+        setResult(`${updatedCalculatorData.total.toString()}  ${updatedCalculatorData.operation}`);
+      }
+    } else if (updatedCalculatorData.next) {
+      setResult(updatedCalculatorData.next.toString());
+    } else if (updatedCalculatorData.total) {
+      setResult(updatedCalculatorData.total.toString());
+    } else {
+      setResult('0');
+    }
+  };
+
+  return (
+    <div>
+      <div className="output-screen">
+        {result}
+      </div>
+      <table className="numbers-container">
+        <tbody>
+          <tr className="numbers">
+            <td>
+              <button type="button" onClick={handleButtonClick}>
+                AC
+              </button>
+            </td>
+            <td>
+              <button type="button" onClick={handleButtonClick}>
+                +/-
+              </button>
+            </td>
+            <td>
+              <button type="button" onClick={handleButtonClick}>
+                %
+              </button>
```

Comment:
- [ ] [OPTIONAL]
It will be good practice to move the buttons into a new component so as to keep the components well organized.